walk-in-interview/src/main/java/com/google/job/data/Location.java

```diff
@@ -0,0 +1,93 @@
+package com.google.job.data;
+
+/** Class that represents the location of a job or an applicant. */
+public final class Location {
+    // TODO(issue/23): Add a postalCode attribute to Location; address is only optional for applicant
+    private final String address;
+    private final String postalCode;
+    private final double latitude;
+    private final double longitude;
+
+    private volatile int hashCode;
+
+    // For serialization
+    public Location() {
+        this(/* address= */"", /* postalCode= */"dummy",
+                /* latitude= */0, /* longitude= */0);
+    }
+
+    public Location(String address, String postalCode, double latitude, double longitude) {
+        // TODO(issue/23): add check for user identity, it is only optional for applicant.
+        this.address = address;
+
+        if (postalCode.isEmpty()) {
```

Comment:
nit: i recommend you do your checks at the end or at the beginning not in the middle of your constructor assignments.

```
this.a = a;
this.b = b;
this.c = c;

// validate a,b,c or throw.

```